src/components/plot/user-controls-grid.tsx

```diff
@@ -33,27 +37,123 @@ const UserControlsGrid = ({
 }: UserControlsGridProps): JSX.Element => {
   const [layout, setLayout] = useState<Layouts>({});
   const [gridProps, setGridProps] = useState<ResponsiveGridProps>({});
+  const [localControlData, setLocalControlData] = useState<any[]>(userControlData.userControlConfig);
+  const [validationComplete, setValidationComplete] = useState<boolean>(false);
 
-  const controlConfig = userControlData.userControlConfig;
+  const alertFunctions = useAlert();
 
   const defaultGridPropsConverted = convertKeysSnakeToCamel(defaultGridProps);
+  // since we're using strict mode, the useEffect is getting called twice in dev mode
+  // this is to guard against it
+  const setupStarted = useRef<boolean>(false);
 
-  // set layout and grid props for local react grid
+  /**
+  * It should be called once to validate the configuration data for the user controls
+  */
   useEffect(() => {
-    if (userControlData?.gridConfig?.layouts && Object.values(userControlData.gridConfig?.layouts).length > 0) {
+    if (setupStarted.current) return;
+    setupStarted.current = true;
+    if (localControlData?.length > 0) {
+
+      //Validate local controls for uid and type. Display error if any control doesn't have an uid or type
+      const invalidControlUid: string[] = [];
+      userControlData?.userControlConfig?.map((control: UserControlConfig) => {
+        if (!control?.uid) {
+          throw new ChaiseError('User Control Error', `Missing UID for the specified global user control  ${control?.label ? `labelled as "${control?.label}"` : ''}`);
+        }
+        if (!control?.type) {
+          invalidControlUid.push(control.uid);
+        }
+      });
+      if (invalidControlUid.length > 0) {
+        alertFunctions.addAlert(`Unable to display the local control with Uid(s) '${invalidControlUid.join(', ')}' as the 'type' was not found for the control`, ChaiseAlertType.ERROR);
+      }
+    }
+
+    //Validate local controls with same uid. Display error in case of same uid(s)
+    const uniqueUid: string[] = [];
+    const dupUid: string[] = [];
+    const controlObject = localControlData.reduce((result, control) => {
+      if (!uniqueUid?.includes(control.uid)) {
+        uniqueUid.push(control.uid)
+      } else {
+        if (!dupUid?.includes(control.uid)) {
+          dupUid.push(control.uid);
+        }
+      }
+      result[control.uid] = control;
+      return result;
+    }, {});
+    if (dupUid?.length > 0) {
+      dupUid.map((id) => alertFunctions.addAlert(`Multiple local controls with the same UID '${id}' were detected`, ChaiseAlertType.ERROR));
+    }
+    userControlData.userControlConfig = Object.values(controlObject);
```

Comment:
We want to avoid modifying the props that are passed to this component. Do we need to update this or is it sufficient to keep the state in this component after validating?

I think it should be sufficient to keep the state in the component since it shouldn't be used outside of it.